src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java

```diff
@@ -0,0 +1,122 @@
+    );
+
+    private static final AtomicAvailability availability = new AtomicAvailability();
+
+    private final String host;
+    private final String path;
+    private final String key;
+    private final boolean isTesting;
+
+    public UsGdotGwinnettTrafficSignalNotifier() {
+        host = PED_SIGNAL_CALL_API_HOST;
+        path = PED_SIGNAL_CALL_API_PATH;
+        key = PED_SIGNAL_CALL_API_KEY;
+        isTesting = false;
+    }
+
+    public UsGdotGwinnettTrafficSignalNotifier(String host, String path, String key) {
+        this.host = host;
+        this.path = path;
+        this.key = key;
+        this.isTesting = true;
+    }
+
+    @Override
+    public void triggerAction(SegmentAction segmentAction, OtpUser otpUser) {
+        String[] idParts = segmentAction.id.split(":");
+        String signalId = idParts[0];
+        String crossingId = idParts[1];
+        triggerPedestrianCall(signalId, crossingId, needsExtendedPhase(otpUser));
+    }
+
+    /** Whether a user needs an extended phase or extra time to cross a signaled intersection. */
+    public static boolean needsExtendedPhase(OtpUser otpUser) {
+        MobilityProfile profile = otpUser.mobilityProfile;
+        if (profile == null) return false;
+
+        String mode = profile.mobilityMode;
+        return mode != null && !mode.equalsIgnoreCase("None");
+    }
+
+    public String getUrl(String signalId, String crossingId, boolean extended) {
+        return host + String.format(path, signalId, crossingId) + (extended ? "?extended=true" : "");
+    }
+
+    public Map<String, String> getHeaders() {
+        return Map.of("X-API-KEY", key);
+    }
+
+    public String getBody() {
+        return "";
+    }
+
+    /**
+     * Trigger a pedestrian call for the given traffic signal and given crossing.
+     * @param signalId The ID of the targeted traffic signal.
+     * @param crossingId The ID of the crossing to activate at the targeted traffic signal.
+     */
+    public boolean triggerPedestrianCall(String signalId, String crossingId, boolean extended) {
+        if (host == null || key == null) {
+            LOG.error("Not triggering pedestrian call: Host and key are not configured.");
+            return false;
+        }
+
+        if (availability.claim()) {
+            try (availability) {
+                if (isTesting) {
+                    // For testing, just wait so that other calls can be attempted.
+                    Thread.sleep(2000);
+                    return true;
+                } else {
+                    String pathAndQuery = getUrl(signalId, crossingId, extended);
+                    var httpResponse = HttpUtils.httpRequestRawResponse(
+                        URI.create(pathAndQuery),
+                        30,
+                        HttpMethod.POST,
+                        getHeaders(),
+                        getBody()
+                    );
+                    if (httpResponse.status == 200) {
```

Comment:
Could replace with HttpStatus.OK_200.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/235/commits/176b5a9acdf5fe2ce09546b71e3673b77470860a.